### ISSUE TITLE
Running coverage against a server

### ISSUE THREAD

**Issue by ORESoftware** [istanbuljs/nyc#610]
I start an Express server with:

`nyc bin/www.js`

the tests are running in a separate process.
When I am done with the tests, I currently manually kill the server with ctrl+c, when I do that, I get this error:

```
 => Server has closed, process is exiting due to SIGINT.
/Users/alexamil/.nvm/versions/node/v6.10.2/lib/node_modules/nyc/node_modules/istanbul-reports/lib/lcovonly/index.js:32
        writer.println('FN:' + [meta.decl.start.line, meta.name].join(','));
                                         ^

TypeError: Cannot read property 'start' of undefined
    at /Users/alexamil/.nvm/versions/node/v6.10.2/lib/node_modules/nyc/node_modules/istanbul-reports/lib/lcovonly/index.js:32:42
    at Array.forEach (native)
    at LcovOnlyReport.onDetail (/Users/alexamil/.nvm/versions/node/v6.10.2/lib/node_modules/nyc/node_modules/istanbul-reports/lib/lcovonly/index.js:30:28)
    at LcovReport.(anonymous function) [as onDetail] (/Users/alexamil/.nvm/versions/node/v6.10.2/lib/node_modules/nyc/node_modules/istanbul-reports/lib/lcov/index.js:21:24)
    at Visitor.(anonymous function) [as onDetail] (/Users/alexamil/.nvm/versions/node/v6.10.2/lib/node_modules/nyc/node_modules/istanbul-lib-report/lib/tree.js:34:30)
    at ReportNode.Node.visit (/Users/alexamil/.nvm/versions/node/v6.10.2/lib/node_modules/nyc/node_modules/istanbul-lib-report/lib/tree.js:123:17)
    at /Users/alexamil/.nvm/versions/node/v6.10.2/lib/node_modules/nyc/node_modules/istanbul-lib-report/lib/tree.js:116:23
    at Array.forEach (native)
    at visitChildren (/Users/alexamil/.nvm/versions/node/v6.10.2/lib/node_modules/nyc/node_modules/istanbul-lib-report/lib/tree.js:115:32)
    at ReportNode.Node.visit (/Users/alexamil/.nvm/versions/node/v6.10.2/lib/node_modules/nyc/node_modules/istanbul-lib-report/lib/tree.js:126:5)

```

**Comment by ORESoftware** [comment]
When I swap out 

`nyc bin/www.js`

and replace it with

`istanbul cover bin/www.js`

then things work, and I get

```
 => Server has closed, process is exiting due to SIGINT.
=============================================================================
Writing coverage object [/Users/alexamil/WebstormProjects/cisco/cdt-now/coverage/coverage.json]
Writing coverage reports at [/Users/alexamil/WebstormProjects/cisco/cdt-now/coverage]
=============================================================================

=============================== Coverage summary ===============================
Statements   : 53.44% ( 1280/2395 )
Branches     : 22.99% ( 186/809 )
Functions    : 31.4% ( 130/414 )
Lines        : 53.65% ( 1280/2386 )
================================================================================

```

**Comment by ciekawy** [comment]
any progress on this? I believe it is same as https://github.com/istanbuljs/nyc/issues/425

**Comment by bcoe** [comment]
@ORESoftware @ciekawy it hasn't been a top priority of the project to get test coverage working in a long lived process, e.g., a server; I would however love to support this.

Would happily help landing any patches that would help address the issues folks are running into -- for starters, I think it would be worthwhile to figure out a very minimal reproduction in an isolated toy repository.

**Comment by ciekawy** [comment]
this is not for sure yet with latest `nyc` and after changing target to ES6 I found no more failures since then. However as this is for my TS project and seem to be related to https://github.com/istanbuljs/nyc/issues/618 now instead of 
```
 => Server has closed, process is exiting due to SIGINT.
/Users/alexamil/.nvm/versions/node/v6.10.2/lib/node_modules/nyc/node_modules/istanbul-reports/lib/lcovonly/index.js:32
        writer.println('FN:' + [meta.decl.start.line, meta.name].join(','));
```
I'm observing periodically (still less frequently) messed coverage reports with reports for lot of the files containing stuff like:
```
All files / server index.ts
60% Statements 6/10 100% Branches 0/0 0% Functions 0/1 100% Lines 6/6
Cannot read property 'start' of undefined
TypeError: Cannot read property 'start' of undefined
    at proj/node_modules/nyc/node_modules/istanbul-reports/lib/html/annotator.js:50:28
    at Array.forEach (native)
    at annotateStatements (proj/node_modules/nyc/node_modules/istanbul-reports/lib/html/annotator.js:46:33)
    at Object.annotateSourceCode (proj/node_modules/nyc/node_modules/istanbul-reports/lib/html/annotator.js:193:9)
    at HtmlReport.onDetail (proj/node_modules/nyc/node_modules/istanbul-reports/lib/html/index.js:217:39)
    at Visitor.(anonymous function) [as onDetail] (proj/node_modules/nyc/node_modules/istanbul-lib-report/lib/tree.js:34:30)
    at ReportNode.Node.visit (proj/node_modules/nyc/node_modules/istanbul-lib-report/lib/tree.js:123:17)
    at proj/node_modules/nyc/node_modules/istanbul-lib-report/lib/tree.js:116:23
    at Array.forEach (native)
    at visitChildren (proj/node_modules/nyc/node_modules/istanbul-lib-report/lib/tree.js:115:32)
```

**Comment by clakech** [comment]
Hi, I have the same issue here @ORESoftware 
Do you find a way to use nyc with a running server ?

**Comment by WittBulter** [comment]
i have a same issue, [here](https://github.com/DhyanaChina/koa-custom-response/blob/master/package.json#L40) is my config file.😯

**Comment by bcoe** [comment]
@WittBulter @clakech definitely a feature we'd like to eventually add to nyc, don't know that there's a simple solution just yet.

**Comment by Morikko** [comment]
I am able to get the coverage with nyc by doing this: `npx nyc node ./www/bin.js`

Note: the only hack I did in the code is to listen to `Ctrl+C` and to suicide the process instead of killing it from the outside with the code:
```javascript
process.on('SIGINT', function() {
    process.exit()
})
```
I discovered this hack thanks to one of our sub-processes that was the only one getting the coverage. Note: I tested this 2 months ago on the version 12. Since, I never tried without the hack on version 13.
